mods/ra/maps/allies-05a/allies05a.lua

```diff
@@ -234,16 +235,52 @@ end
 
 InitTriggers = function()
 	Trigger.OnInfiltrated(Warfactory, function()
+		if greece.IsObjectiveCompleted(infWarfactory) then
+			return
+		elseif Truk.IsDead then
+			if not greece.IsObjectiveCompleted(mainObj) then
+				ussr.MarkCompletedObjective(ussrObj)
+			end
+
+			return
+		end
+
 		Trigger.ClearAll(Spy)
 		greece.MarkCompletedObjective(infWarfactory)
 		WarfactoryInfiltrated()
 	end)
 
+	Trigger.OnKilled(Truk, function()
+		if not greece.IsObjectiveCompleted(infWarfactory) then
+			greece.MarkFailedObjective(infWarfactory)
+		elseif FollowTruk then
+			ussr.MarkCompletedObjective(ussrObj)
+		end
+	end)
+
 	Trigger.OnInfiltrated(Prison, function()
+		if greece.IsObjectiveCompleted(mainObj) then
+			return
+		end
+
 		if not greece.IsObjectiveCompleted(infWarfactory) then
 			Media.DisplayMessage("Good work! But next time skip the heroics!", "Battlefield Control")
 			greece.MarkCompletedObjective(infWarfactory)
```

Comment:
Shouldn't this be `MarkFailedObjective` as you didn't and can no longer infiltrate the war factory?

Reply:
In theory yes, but in practice I disagree. The warfactory infiltration objective is only supporting, not required, so imho it makes sense to allow a flawless playthrough when you chose the harder route.